### DISCUSSION TITLE
Get usage by timestamp

### DISCUSSION THREAD

**Discussion by harsh020** [deepgram/community#864] [General help] [answered]
Hi guys, 

Are there any APIs that can be used to fetch the billing/usage based on timestamp? I mean not just date but date + time range (i.e start -> end). For my use-case, I want to keep track of the usage per usage. 

I tried inspecting the api calls at the billing page but couldn't find any such api calls that could be used with range and api-key. 

P.S: I know we can calculate using the character count/ time at our end. But wanted to check if there is something like this available because we are not maining a db to store the details at our end.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment, marked as answer]
The ["Get All Requests"](https://developers.deepgram.com/reference/get-all-requests) API from Deepgram allows you to specify a date and time range for fetching usage data. The start and end dates of the requested date range can be specified in the formats YYYY-MM-DD, YYYY-MM-DDTHH:MM:SS, or YYYY-MM-DDTHH:MM:SS+HH:MM. This means you can specify both the date and time for the start and end of the range.

**Reply by harsh020** [reply]
Thank you very much @SandraRodgers!
